examples/graph_cross_network.py

```diff
@@ -0,0 +1,248 @@
+                d = degree(g.edge_index.flatten(), num_nodes, dtype=torch.long)
+            else:
+                d = degree(g.edge_index[1, :], num_nodes, dtype=torch.long)
+            node_feat[torch.arange(num_nodes), d - min_degree] = 1.
+            if g.x is not None:
+                g.x = torch.cat((g.x, node_feat), dim=1)
+            else:
+                g.x = node_feat
+
+    def augment_self_loops(self):
+        for i in range(len(self.graphs)):
+            edge_index = add_remaining_self_loops(self.graphs[i].edge_index)
+            self.graphs[i].edge_index = edge_index[0]
+
+    def get_min_max_degrees(self):
+        min_degree = sys.maxsize
+        max_degree = 0
+        for g in self.graphs:
+            if g.is_undirected():
+                degrees = degree(g.edge_index.flatten()).long()
+            else:
+                degrees = degree(g.edge_index[1, :]).long()
+            min_degree = min(min_degree, degrees.min().item())
+            max_degree = max(max_degree, degrees.max().item())
+        return min_degree, max_degree
+
+    def len(self):
+        return len(self.graphs)
+
+    def get(self, idx):
+        return self.graphs[idx]
+
+    def statistics(self):
+        return self.graphs[0].x.shape[1], self.num_classes
+
+
+class GraphClassifier(torch.nn.Module):
+    """
+    Description
+    -----------
+    Graph Classifier for graph classification.
+    GXN + MLP
+    """
+    def __init__(self, args):
+        super(GraphClassifier, self).__init__()
+        self.gxn = GraphCrossNet(in_dim=args.in_dim,
+                                 out_dim=args.embed_dim,
+                                 edge_feat_dim=args.edge_feat_dim,
+                                 hidden_dim=args.hidden_dim,
+                                 pool_ratios=args.pool_ratios,
+                                 readout_nodes=args.readout_nodes,
+                                 conv1d_dims=args.conv1d_dims,
+                                 conv1d_kws=args.conv1d_kws,
+                                 cross_weight=args.cross_weight,
+                                 fuse_weight=args.fuse_weight)
+        self.lin1 = torch.nn.Linear(args.embed_dim, args.final_hidden_dim)
+        self.lin2 = torch.nn.Linear(args.final_hidden_dim, args.out_dim)
+        self.dropout = args.dropout
+
+    def forward(self, graph):
+        embed, logits1, logits2 = self.gxn(graph)
+        logits = F.relu(self.lin1(embed))
+        if self.dropout > 0:
+            logits = F.dropout(logits, p=self.dropout, training=self.training)
+        logits = self.lin2(logits)
+        return F.log_softmax(logits, dim=1), logits1, logits2
+
+
+def compute_loss(cls_logits: Tensor, labels: Tensor, logits_s1: Tensor,
+                 logits_s2: Tensor, epoch: int, total_epochs: int,
+                 device: torch.device):
+    # classification loss
+    classify_loss = F.nll_loss(cls_logits, labels.to(device))
+
+    # loss for vertex infomax pooling
```

Comment:
Can we integrate the computation of the VIX loss into `GraphCrossNet` model, *e.g.*, `GraphCrossNet.pooling_loss(...)`?